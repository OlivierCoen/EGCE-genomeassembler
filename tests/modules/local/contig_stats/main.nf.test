nextflow_process {

    name "Test Process CONTIG_STATS"
    script "modules/local/contig_stats/main.nf"
    process "CONTIG_STATS"
    tag "contig_stats"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] =  Channel.of(
                    [
                        [ id: "test" ],
                        file( '$projectDir/tests/test_data/contig_stats/*.fasta', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
