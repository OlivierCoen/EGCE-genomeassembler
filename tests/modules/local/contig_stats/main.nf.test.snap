{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_contig_sizes.tsv:md5,6df5187b2f854172448e4a8887b896b0"
                    ]
                ],
                "1": [
                    [
                        "CONTIG_STATS",
                        "python",
                        "3.13.3"
                    ]
                ],
                "2": [
                    [
                        "CONTIG_STATS",
                        "pandas",
                        "2.3.0"
                    ]
                ],
                "3": [
                    [
                        "CONTIG_STATS",
                        "biopython",
                        "1.85"
                    ]
                ],
                "contig_sizes": [
                    [
                        {
                            "id": "test"
                        },
                        "test_contig_sizes.tsv:md5,6df5187b2f854172448e4a8887b896b0"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T11:01:00.87489233"
    }
}